### DISCUSSION TITLE
Break batch embeddings into each sample

### DISCUSSION THREAD

**Discussion by zcaicaros** [pyg-team/pytorch_geometric#2465] [Q&A] [answered]
Hi,
How can I break the batch node embeddings? After processing  a `data.Batch` batch of data, the node embedding has dimension `[batch_size * #nodes_in_each_graph, output_dim]`, but how can I break it into `[batch_size, nodes_#in_each_graph, output_dim]`?

**Comment by rusty1s** [comment, marked as answer]
You can use `torch_geometric.utils.to_dense_batch` to achieve that. Note that for graphs with varying number of nodes, it will return a tensor of shape `[batch_size, max_num_nodes, output_dim]`, filled with zeros for smaller graphs.
